pkg/backend/httpstate/client/api_endpoints.go

```diff
@@ -115,4 +115,8 @@ func init() {
 
 	// APIs for managing `PolicyPack`s.
 	addEndpoint("POST", "/api/orgs/{orgName}/policypacks", "publishPolicyPack")
+
+	// APIs for managing Search capabilities
+	addEndpoint("GET", "/api/orgs/{orgName}/search/resources?source=cli", "getSearchResources")
+	addEndpoint("GET", "/api/orgs/{orgName}/search/resources/parse?source=cli", "getSearchResourcesParse")
```

Comment:
```suggestion
	addEndpoint("GET", "/api/orgs/{orgName}/search/resources", "getSearchResources")
	addEndpoint("GET", "/api/orgs/{orgName}/search/resources/parse", "getSearchResourcesParse")
```

AFAIK, these are only used for tracing to get friendlier names for a given method & path. So the extra `?source=cli` isn't going to be sent to the server, and will actually impede it from getting the friendlier `getSearchResources` or `getSearchResourcesParse` names.

With these as:

```go
addEndpoint("GET", "/api/orgs/{orgName}/search/resources?source=cli", "getSearchResources")
addEndpoint("GET", "/api/orgs/{orgName}/search/resources/parse?source=cli", "getSearchResourcesParse")
```

this one-off test fails:

```go
func TestTracingEndpointName(t *testing.T) {
	path := fmt.Sprintf("/api/orgs/%s/search/resources", url.PathEscape("myorg"))
	name := getEndpointName(http.MethodGet, path)
	assert.Equal(t, "api/getSearchResources", name)
}
```

```
Error:      	Not equal:
            	expected: "api/getSearchResources"
            	actual  : "unknown"
```

After removing the `?source=cli`, the test succeeds.

If you want to pass along the `?source=cli`, I think you'll need to do that with the query param object passed to `pc.restCall` in `client.go`.

Just be sure not to add it to the path itself, because if there is a query object, it'll try to add a `?`:

https://github.com/pulumi/pulumi/blob/93c70d82ccd6d677a4fcab32a2a97bc05ded1575/pkg/backend/httpstate/client/api.go#L379-L390

Reply:
Let's not block on this, especially if we're only enabling this as a dev experiment for now.

@glena has suggested using headers for this purpose instead, which I agree is a better long-term strategy. I think we're still plumbing that together on the service side, so it would be OK to punt on this until the service is ready.

Reply:
yep go ahead with that approach, once my changes I can take care of updating these